transformer_engine/paddle/layer/base.py

```diff
@@ -280,6 +338,21 @@ def grad_output_preprocess(
 
         return grad_output_mat, grad_output_c, grad_output_t, bgrad
 
+    def set_nccl_overlap_warning_if_tp(self) -> None:
+        """When using TP, the NCCL communication needs to be scheduled
+        before the GEMM for there to be a guaranteed overlap. From the
+        host side in TE, the comm calls are always launched first, but
```

Comment:
Although the NCCL is always invoked first, to ensure that the GEMM isn't scheduled first, the environment variable `CUDA_DEVICE_MAX_CONNECTIONS` needs to be set to 1 to force using a single channel.

Reply:
Is it a PaddlePaddle limitation?

Reply:
Same as `[here](https://github.com/NVIDIA/TransformerEngine/blob/66ff2e36ec79e286ec6c6989db8eecc808ca895c/transformer_engine/pytorch/module/base.py#L644)`. Need to check if such limitation is necessary for Paddle.

Reply:
Need to think about replacing `CUDA_DEVICE_MAX_CONNECTIONS` with `cudaEventWait` if overlapping setup is needed.